### DISCUSSION TITLE
How to run tests against different local nuget packages with dynamic versions?

### DISCUSSION THREAD

**Discussion by egil** [dotnet/BenchmarkDotNet#2437] [Q&A]
During a CI build (e.g. PR), I would like to run a few benchmarks against different versions of a library (last released, main-branch, pr-branch).

My rough plan is this:

1. Check out each branch/tag that represents each version and use `dotnet pack` to build a Nuget package for that version, and store the created packages in a local directory on the CI machine.
2. Start the benchmark and somehow pass in the version numbers of the three Nuget packages to three Job definitions. The benchmark project will have a nuget.config file with a pointer to the local package directory.
3. Collect output and share in comments on PR.

It is the second step I am not sure how to do. I have a bunch of ideas, but they all seem slightly hacky. What is the recommended approach?

**Comment by timcassell** [comment]
You can use `Job.WithNuget` extension.

```cs
var config = DefaultConfig.Instance.AddJob(Job.Default.WithNuGet("MyPackage", "1.0.0"));
```

**Reply by egil** [reply]
Looks about right. I'll give it a try.
